server.js

```diff
@@ -21,6 +21,71 @@ app.get("/", (req, res) => {
   res.send("Hello Technigo!");
 });
 
+//////// Live sesh //////////
+const { Schema } = mongoose;
+const ThoughtSchema = new Schema ({
+  message: {
+    // Most important one is:
+    type: String,
+    // required is true or false
+    required: true,
+    minlength: 4,
+    maxlength: 140,
+    //removes unnecessary white spaces from string
+    trim: true
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+  })
+
+const Thought = mongoose.model("Thought", ThoughtSchema)
+
+app.get('/thoughts', async (req, res) => {
+  const displayThoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec();
+    res.json(displayThoughts)
+})
+
+app.post('/thoughts', async (req, res) => {
+  const { message, createdAt } = req.body;
+  try {
+    const savedThought = await new Thought({ message, createdAt }).save();
+    res.status(201).json({
+      success: true, 
+      message: "Posted!", 
+      response: savedThought
+    });
+  }
+  catch (e){
+    res.status(400).json({
+      success: false, 
+      response: e,
+      message: 'Could not save thought to database'
+    });
+  }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params;
+  try {
+   const updateLikes = await Thought.findByIdAndUpdate(thoughtId, { $inc: { likes: 1 }});
```

Comment:
in this snippet you increase "likes" but in the thoughts schema the likes are represented by "hearts", which is probably why this endpoint doesn't work. this is my code for this endpoint:

      const singleThought = await Thought.findById(req.params._id);
      if (singleThought) {
        singleThought.hearts += 1;
        await singleThought.save();